apps/api/src/services/account-balance/account-balance.service.ts

```diff
@@ -13,4 +14,18 @@ export class AccountBalanceService {
       data
     });
   }
+
+  public async getAccountBalances(accountId: string): Promise<AccountBalances> {
+    const balances = await this.prismaService.accountBalance.findMany({
+      where: {
+        accountId: accountId,
```

Comment:
Change to
```
accountId,
userId
```

This ensures that you cannot get the balances of a foreign account by guessing its id.